File: server/app/models/product.py
from app.extentions import db

class Product(db.Model):
    __tablename__ = "products"

    id = db.Column(db.Integer, primary_key=True)
    name = db.Column(db.String(120), nullable=False)
    price = db.Column(db.Float, nullable=False)
    image_url = db.Column(db.String(255))
    description = db.Column(db.Text)
    category = db.Column(db.String(50))
    is_best_seller = db.Column(db.Boolean)


    def to_dict(self):
        return {
            "id": self.id,
            "name": self.name,
            "price": self.price,
            "image_url": self.image_url,
            "description": self.description,
            "category": self.category,
            "is_best_seller": self.is_best_seller
        }

File: src/types/product.ts
export interface Review {
    id: number;
    rating: number;
    comment: string;
    username: string;
    user_id: number;
    created_at: string; // ISO string from Flask
}

export interface Product {
    id: number;
    name: string;
    price: number;
    images: string[];
    category: string;
    stock: number;
    image_url: string;
    is_best_seller?: boolean;
    description: string;
    reviews: Review[]; // Nested array from your Product.to_dict()
    average_rating: number;
}

export interface ShopContextType {
    products: Product[];
    isLoading: boolean;
    isError: boolean;
    search: string;
    setSearch: (value: string) => void;
    showSearch: boolean;
    setShowSearch: (value: boolean) => void;
    selectedCategories: string[]; 
    setSelectedCategories: (categories: string[]) => void;
    toggleCategory: (value: string) => void;
}